syntax = "proto3";
package role;





/*
GetRoleDetail请求
*/
message GetRoleDetailRequest {
    /*
    角色Id
    */
    string id = 1 ;
}

/*
GetRoleDetail返回
*/
message GetRoleDetailResponse {
    /*
    角色禁用菜单列表
    */
    repeated string forbidden_menu = 1 ;
    /*
    角色id
    */
    string id = 2 ;
    /*
    角色所在org
    */
    int32 org = 3 ;
    /*
    角色名称
    */
    string role = 4 ;
    /*
    角色下用户列表
    */
    repeated string user = 5 ;
    /*
    角色下权限id列表
    */
    repeated string permission = 6 ;
    /*
    角色下用户组列表
    */
    repeated string user_group = 7 ;
}

/*
GetRoleDetailApi返回
*/
message GetRoleDetailResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    role.GetRoleDetailResponse data = 4 ;
}

